src/main/java/io/jenkins/plugins/intotorecorder/transport/GrafeasTransport.java

```diff
@@ -0,0 +1,190 @@
+/**
+ *
+ */
+package io.jenkins.plugins.intotorecorder.transport;
+
+import io.github.in_toto.models.Link;
+import io.github.in_toto.models.Artifact.ArtifactHash;
+import io.github.in_toto.keys.Signature;
+import java.net.URI;
+import org.apache.http.client.utils.URIBuilder;
+import java.net.URISyntaxException;
+import java.util.*;
+import com.google.gson.Gson;
+
+import java.io.IOException;
+
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.ByteArrayContent;
+import com.google.api.client.http.GenericUrl;
+
+
+public class GrafeasTransport extends Transport {
+
+    URI uri;
+    GrafeasOccurrence occurrence;
+
+    public static class GrafeasInTotoMetadata {
+        // This class exists to represent the signed document format for Grafeas
+        // in-toto links.
+
+        @edu.umd.cs.findbugs.annotations.SuppressWarnings("URF_UNREAD_FIELD")
+        private ArrayList<Signature> signatures = new ArrayList<Signature>();
+        @edu.umd.cs.findbugs.annotations.SuppressWarnings("URF_UNREAD_FIELD")
+        private GrafeasInTotoLink signed;
+
+        private static class GrafeasInTotoLink {
+            // This class exists to represent the Grafeas document format for
+            // in-toto links.
+
+            private static class Artifact {
+                @edu.umd.cs.findbugs.annotations.SuppressWarnings("URF_UNREAD_FIELD")
+                private String resourceUri;
+                @edu.umd.cs.findbugs.annotations.SuppressWarnings("URF_UNREAD_FIELD")
+                private Map<String, String> hashes;
+
+                private Artifact(String resourceUri, Map<String, String> hashes) {
+                    this.resourceUri = resourceUri;
+                    this.hashes = hashes;
+                }
+
+            }
+
+            @edu.umd.cs.findbugs.annotations.SuppressWarnings("URF_UNREAD_FIELD")
+            private List<String> command = new ArrayList<String>();
+            @edu.umd.cs.findbugs.annotations.SuppressWarnings("URF_UNREAD_FIELD")
+            private List<Artifact> materials = new ArrayList<Artifact>();
+            @edu.umd.cs.findbugs.annotations.SuppressWarnings("URF_UNREAD_FIELD")
+            private List<Artifact> products = new ArrayList<Artifact>();
+            private Map<String, Map<String, String>> byproducts = new HashMap<String, Map<String, String>>();
+            private Map<String, Map<String, String>> environment = new HashMap<String, Map<String, String>>();
+
+            public GrafeasInTotoLink(List<String> command,
```

Comment:
part of me wonders if we want to ticketize making this a little bit more exportable, so as to provide a blessed way to do this transformation in Java to other users.

Reply:
Done in #16.